torch/distributions/half_cauchy.py

```diff
@@ -0,0 +1,61 @@
+import math
+
+from torch.distributions import constraints
+from torch.distributions.transforms import AbsTransform
+from torch.distributions.cauchy import Cauchy
+from torch.distributions.transformed_distribution import TransformedDistribution
+
+
+class HalfCauchy(TransformedDistribution):
+    r"""
+    Creates a half-normal distribution parameterized by `scale` where::
+
+        X ~ Cauchy(0, scale)
+        Y = |X| ~ HalfCauchy(scale)
+
+    Example::
+
+        >>> m = HalfCauchy(torch.tensor([1.0]))
+        >>> m.sample()  # half-cauchy distributed with scale=1
+        tensor([ 2.3214])
+
+    Args:
+        scale (float or Tensor): scale of the full Cauchy distribution
+    """
+    arg_constraints = {'scale': constraints.positive}
+    support = constraints.positive
+    has_rsample = True
+
+    def __init__(self, scale, validate_args=None):
+        super(HalfCauchy, self).__init__(Cauchy(0, scale), AbsTransform(),
+                                         validate_args=validate_args)
+
+    @property
+    def scale(self):
+        return self.base_dist.scale
+
+    @property
+    def mean(self):
+        return self.base_dist.mean
+
+    @property
+    def variance(self):
+        return self.base_dist.variance
+
+    def log_prob(self, value):
+        if self._validate_args:
+            self._validate_sample(value)
```

Comment:
Can you confirm that we want to do double validation here? (since `base_dist.log_prob` will also validate the arg)

Reply:
Yeah, I thought that we need to check the support but it is unnecessary because base_dist also checks that.